### PR TITLE
Add a run decorator

### PR DESCRIPTION
**Pull request by jan-janssen**
Based on https://github.com/pyiron/pyiron_base/pull/987

### CODE REVIEW

**Comment by jan-janssen** [comment]
It is a bit surprising that the worker job fails with these changes.

**Comment by samwaseda** [comment]
Can you add doc string describing what this is used for? What it does is relatively clear, but in the pyiron context it's not straightforward to see when it should be applied (or if it's only meant to be used in `GenericJob` then I think it can also be a private function).

File: pyiron_base/jobs/job/runfunction.py [hunk 1]
```diff
@@ -456,6 +454,19 @@ def run_job_with_runmode_srun(job):
     )
 
 
+def run_static_decorator(func):
+    def wrapper(job):
+        if job.job_id is not None:
+            job.project.db.item_update({"timestart": datetime.now()}, job.job_id)
```
Comment (samwaseda) [inline pyiron_base/jobs/job/runfunction.py:461]: Based on [my comment](https://github.com/pyiron/pyiron_base/issues/979#issuecomment-1384891292), I would suggest to write:

```python
update_dict = {"timestop": None, "totalcputime": None}
if job.project.db.get_item_by_id(job._job_id)["timestart"] is None:
    update_dict["timestart"] = datetime.now()
job.project.db.item_update(update_dict, job.job_id)
```

File: pyiron_base/jobs/job/runfunction.py [hunk 2]
```diff
@@ -456,6 +454,19 @@ def run_job_with_runmode_srun(job):
     )
 
 
+def run_static_decorator(func):
```
Comment (pmrv) [inline pyiron_base/jobs/job/runfunction.py]: The name is not very clear, maybe `set_runtime` or something like that.

File: pyiron_base/jobs/job/runfunction.py (hunk 1 above)
Comment (jan-janssen) [reply pyiron_base/jobs/job/runfunction.py:461]: > Based on [my comment](https://github.com/pyiron/pyiron_base/issues/979#issuecomment-1384891292), I would suggest to write:
> 
> ```python
> update_dict = {"timestop": None, "totalcputime": None}
> if job.project.db.get_item_by_id(job._job_id)["timestart"] is None:
>     update_dict["timestart"] = datetime.now()
> job.project.db.item_update(update_dict, job.job_id)
> ```

Running `run_static()` multiple times currently only works for the `GenericMasters` including the `InteractiveWrapper`. For a regular job this is something we do not support at the moment, for example for VASP we would overwrite the output, so to reduce the overhead I would not add another database request to get the `timestart` currently stored in the database.

File: pyiron_base/jobs/job/runfunction.py (hunk 2 above)
Comment (jan-janssen) [reply pyiron_base/jobs/job/runfunction.py]: I renamed the decorator to `run_time_decorator`.

**Pull request merged**